api/src/main/java/com/code4ro/nextdoor/security/jwt/JwtTokenProvider.java

```diff
@@ -15,25 +15,40 @@
 public class JwtTokenProvider {
     private static final Logger LOGGER = LoggerFactory.getLogger(JwtTokenProvider.class);
 
-    @Value("${app.jwtSecret}")
+    @Value("${app.jwt.secret}")
     private String jwtSecret;
 
-    @Value("${app.jwtExpirationInMs}")
-    private int jwtExpirationInMs;
+    @Value("${app.jwt.accessTokenExpirationInMs}")
+    private long accessTokenExpirationInMs;
 
-    public String generateToken(final Authentication authentication) {
-
-        final UserPrincipal userPrincipal = (UserPrincipal) authentication.getPrincipal();
+    @Value("${app.jwt.refreshTokenExpirationInMs}")
+    private long refreshTokenExpirationInMs;
 
+    public String generateAccessToken(final UUID userId) {
         final Date now = new Date();
-        final Date expiryDate = new Date(now.getTime() + jwtExpirationInMs);
+        final Date expiryDate = new Date(now.getTime() + accessTokenExpirationInMs);
 
         return Jwts.builder()
-                .setSubject(userPrincipal.getId().toString())
+                .setSubject(userId.toString())
```

Comment:
The safest way to sign the access token is using Private key and implementing RSA algorithm
Also, for the current implementation the signing key is too short